sips/sip-028/sip-028-sbtc_peg.md

```diff
@@ -0,0 +1,217 @@
+
+## Proposed Solution
+
+sBTC aims to mitigate Bitcoin’s limitations by combining the capability of the Stacks Blockchain with the reliability and security of Bitcoin. By enabling the secure movement of BTC in and out of the Stacks Blockchain via the sBTC protocol, users can interact with their BTC on Stacks using Clarity smart contracts and fast block times. The protocol is “secure” in that it is operated by a decentralized signer network, removing the risk of a single point of failure and trust in a single custodian. Users can deposit BTC into the protocol, seamlessly transact using sBTC on the Stacks blockchain, and have the freedom to redeem sBTC tokens for the underlying BTC at any time.
+
+### Programmability
+
+[Clarity](https://docs.stacks.co/clarity/overview) is the smart contract language on Stacks, which allows developers to encode essential business logic on a blockchain. Using smart contracts, developers can build more expressive decentralized applications that interact with sBTC, such as DeFi protocols, stablecoins, payments, and many others.
+
+### Fast Blocks
+
+The Stacks Nakamoto Upgrade, proposed in [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md#proposed-solution), enables fast blocks where user-submitted transactions will now take on the order of seconds, instead of tens of minutes. Thus, sBTC on Stacks Nakamoto will offer an improvement to Bitcoin’s current transaction times.
+
+The sBTC protocol not only addresses the limitations of the Bitcoin scripting system but also provides a secure and decentralized solution for utilizing Bitcoin in various applications.
+
+## Design
+
+While the first sBTC implementation is under development, the wrapped nature of the sBTC token means that any such system would have the following properties:
+
+- sBTC is a SIP-10 token backed 1:1 by BTC.
+- The sBTC peg wallet is maintained by the set of sBTC signers. These signers are responsible for the security and maintenance of the wallet, ensuring that sBTC is redeemable for BTC.
+- Bitcoin can be converted into sBTC within 3 Bitcoin blocks; and sBTC can be converted into Bitcoin within 6 Bitcoin blocks. sBTC relies on the forking behavior guaranteed by [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md) in order to maintain the peg wallet correctly across forks.
```

Comment:
Nit: the use of the semicolon is grammatically incorrect.  Use a comma please.

Reply:
> sBTC relies on the forking behavior guaranteed by [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md) in order to maintain the peg wallet correctly across forks.

I'm a little confused by this.  Can you specifically state the forking rules you're relying on?  Asking because while the first production release of Nakamoto guarantees that each Bitcoin fork has at most one Stacks fork after two confirmations, it doesn't guarantee anything about the confirmation of transactions which land on a Stacks fork whose Bitcoin fork gets orphaned.  The SIP states that this will be addressed after the release.  I just want to confirm that sBTC does not depend on this particular behavior?  Because if it does, then this SIP would be blocked on that, and it should be stated up-front.

Reply:
@aldur

Reply:
This does not refer to transaction replay, which is outside of the scope of the SIP. It is referring to the fact that Stacks blockchain will no longer fork on its own, it will always follow Bitcoin forks.